creature.cpp

```diff
@@ -1144,6 +1144,8 @@ static string getAttackParam(AttackType type) {
     case AttackType::SPELL: return "spell";
     case AttackType::POSSESS: return "touch";
   }
+  assert("__FUNCTION__ case unknown attack type");
+  return "";
```

Comment:
clang -v
Ubuntu clang version 3.5.0-4ubuntu2 (tags/RELEASE_350/final) (based on LLVM
3.5.0)
Target: x86_64-pc-linux-gnu
Thread model: posix

FAIL fails exactly the way I want (at the moment causes a segfault which is
nicely caught in stack_printer.cpp). It also creates a temporary stream, so
I can do FAIL << variable << " " << otherVariable

I don't use assert anywhere in the code, I think.

On Fri, Apr 24, 2015 at 4:45 PM, moozaad <notifications@github.com> wrote:

> In creature.cpp
> <https://github.com/miki151/keeperrl/pull/674#discussion_r29054248>:
>
> > @@ -1144,6 +1144,8 @@ static string getAttackParam(AttackType type) {
> >      case AttackType::SPELL: return "spell";
> >      case AttackType::POSSESS: return "touch";
> >    }
> > +  assert("__FUNCTION__ case unknown attack type");
> > +  return "";
>
> True, tbh I forgot C++ did that with case and enums. I just had to go
> double check the gcc man. I was mainly focused on palliatively cleaning out
> the compiler warnings without turning too many off as in my experience,
> that's always a bad idea.
>
> What version of clang are you using as it doesn't compile on 3.5 here.
>
> Just curious, why the custom FAIL instead of the assert?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/miki151/keeperrl/pull/674/files#r29054248>.
>